drivers/ethernet/eth_xlnx_gem.c

```diff
@@ -225,21 +217,22 @@ static void eth_xlnx_gem_iface_init(struct net_if *iface)
 	const struct device *dev = net_if_get_device(iface);
 	const struct eth_xlnx_gem_dev_cfg *dev_conf = dev->config;
 	struct eth_xlnx_gem_dev_data *dev_data = dev->data;
+#ifdef CONFIG_MDIO
+	struct phy_link_state state = {};
+#endif
 
 	/* Set the initial contents of the current instance's run-time data */
 	dev_data->iface = iface;
 	net_if_set_link_addr(iface, dev_data->mac_addr, 6, NET_LINK_ETHERNET);
 	ethernet_init(iface);
-	net_if_carrier_off(iface);
+	net_eth_carrier_off(iface);
```

Comment:
consider making the ``phy-handle`` required in the ``xlnx,gem.yaml``. It seems to be needed to configure the GEM Network Configuration Register and clock speed. also without a phy the carrier would never be set to on.